Mathlib/MeasureTheory/Function/UnifTight.lean

```diff
@@ -0,0 +1,514 @@
+        meas_ge_le_lintegral_div hmf.aemeasurable (by norm_num) (by norm_num)
+    _ < ∞ := by apply div_lt_top hf.ne (by norm_num)
+  set s := { x | 1 / (↑M + 1) ≤ f x }
+  -- fulfill the goal
+  use s, hms, hμs, hM
+
+/-- A single function that is `Memℒp f p μ` is tight wrt to `μ`. -/
+theorem Memℒp.snorm_indicator_compl_le (hp_one : 1 ≤ p) (hp_top : p ≠ ∞)
+    {f : α → β} (hf : Memℒp f p μ)
+    {ε : ℝ} (hε : 0 < ε) :
+    ∃ (s : Set α) (_ : MeasurableSet s) (_ : μ s < ∞),
+      snorm (sᶜ.indicator f) p μ ≤ ENNReal.ofReal ε := by
+  -- The proof unwraps `Memℒp f p μ` and applies the analogous result for `lintegral`.
+  -- do some arithmetic that will come in useful
+  have hp_pos := zero_lt_one.trans_le hp_one
+  have hp_nz := hp_pos.ne'
+  have hrp_pos : 0 < p.toReal := ENNReal.toReal_pos hp_nz hp_top
+  have hεp : 0 < ε ^ p.toReal := by simp only [Real.rpow_pos_of_pos, hε]
+  -- decode Memℒp into a.e. strong measurability and finite snorm
+  obtain ⟨haesmf, hsnf⟩ := hf
+  -- transform snorm to lintegral
+  rw [snorm_eq_snorm' (by assumption) (by assumption)] at hsnf
+  have hinpf := calc
+    ∫⁻ a, ‖f a‖₊ ^ p.toReal ∂μ
+      = (snorm' f p.toReal μ) ^ p.toReal := lintegral_rpow_nnnorm_eq_rpow_snorm' hrp_pos
+    _ < ∞                                := (rpow_lt_top_iff_of_pos hrp_pos).mpr hsnf
+  -- get a.e. measurability for the integrand
+  -- XXX: Why does `AEStronglyMeasurable.ennnorm` only give the weaker AEMeasurable?
+  --      It would make sense to me to use `haesmf.ennnorm.aemeasurable` below.
+  have haemnf := haesmf.ennnorm
+  have haemnpf := haemnf.pow_const p.toReal
+  -- use core result for lintegral (needs only AEMeasurable), the target estimate will be in `hsfε`
+  obtain ⟨s, hms, hμs, hsfε⟩ := lintegral_indicator_compl_le haemnpf hinpf hεp
+  use s, hms, hμs
+  -- move indicator through function compositions, XXX: is this simp-able?
+  rw [← Function.comp_def (fun x : ℝ≥0∞ => x ^ p.toReal)] at hsfε
+  rw [← Function.comp_def ENNReal.ofNNReal] at hsfε
+  rw [← Function.comp_def nnnorm] at hsfε
+  rw [sᶜ.indicator_comp_of_zero (@ENNReal.zero_rpow_of_pos p.toReal hrp_pos)] at hsfε
+  rw [sᶜ.indicator_comp_of_zero ENNReal.coe_zero] at hsfε
+  rw [sᶜ.indicator_comp_of_zero nnnorm_zero] at hsfε
+  rw [Function.comp_def nnnorm] at hsfε
+  rw [Function.comp_def ENNReal.ofNNReal] at hsfε
+  rw [Function.comp_def (fun x : ℝ≥0∞ => x ^ p.toReal)] at hsfε
+  -- convert lintegral to snorm
+  rw [lintegral_rpow_nnnorm_eq_rpow_snorm' hrp_pos] at hsfε
+  rw [← snorm_eq_snorm' (by assumption) (by assumption)] at hsfε
+  -- commute ENNReal coersion with rpow, use rpow monotonicity
+  rw [← ofReal_rpow_of_pos (by assumption)] at hsfε
+  rw [ENNReal.rpow_le_rpow_iff hrp_pos] at hsfε
+  exact hsfε
+
+/-- A constant function is tight. -/
+theorem unifTight_const {g : α → β} (hp : 1 ≤ p) (hp_ne_top : p ≠ ∞) (hg : Memℒp g p μ) :
+    UnifTight (fun _ : ι => g) p μ := fun ε hε ↦ by
+  by_cases hε_top : ε = ∞
```

Comment:
Let's enter tactic mode directly. It looks cleaner than the `fun ε hε ↦ by` way you have here:
```suggestion
    UnifTight (fun _ : ι => g) p μ := by
  intro ε hε
  by_cases hε_top : ε = ∞
```

Reply:
I was following Yury's lead with this style. :-) My original code use `intro` everywhere. I myself have no strong opinion here.